src/main/java/school/hei/patrimoine/visualisation/swing/ihm/google/GoogleDocsLinkVerfierScreen.java

```diff
@@ -0,0 +1,214 @@
+    returnButton.addActionListener(returnToPreviousScreen());
+    return returnButton;
+  }
+
+  private void addInputFieldsFromData(List<NamedString> linksData) {
+    GridBagConstraints gbc = new GridBagConstraints();
+    gbc.gridx = 0;
+    gbc.fill = HORIZONTAL;
+    gbc.insets = new Insets(10, 50, 10, 50);
+
+    int yPosition = 2;
+    for (NamedString linkData : linksData) {
+      var nameLabel = new JLabel(linkData.name());
+      nameLabel.setFont(new Font("Arial", BOLD, 18));
+      nameLabel.setHorizontalAlignment(LEFT);
+
+      gbc.gridy = yPosition++;
+      inputPanel.add(nameLabel, gbc);
+
+      JTextField newField = newGoogleDocsLinkTextField(linkData.value());
+      inputFields.add(newField);
+
+      JScrollPane scrollPane = new JScrollPane(newField);
+      gbc.gridy = yPosition++;
+      inputPanel.add(scrollPane, gbc);
+    }
+  }
+
+  private JTextField newGoogleDocsLinkTextField(String initialValue) {
+    var textField = new JTextField(70);
+    textField.setInputVerifier(linkIdInputVerifier);
+    textField.setFont(new Font("Arial", PLAIN, 16));
+    textField.setText(initialValue);
+    linkIdInputVerifier.verify(textField);
+    return textField;
+  }
+
+  private ActionListener returnToPreviousScreen() {
+    return e -> {
+      invokeLater(() -> new GoogleDocsSubmitScreen(googleApi, authDetails));
+      inputFrame.setVisible(false);
+    };
+  }
+
+  private void loadDataInBackground(List<NamedString> linksData) {
+    var loadingDialog = new JDialog(inputFrame, "Processing", true);
+    var loadingLabel = new JLabel("Processing, please wait...");
+    loadingLabel.setHorizontalAlignment(SwingConstants.CENTER);
+    loadingDialog.getContentPane().add(loadingLabel, CENTER);
+    loadingDialog.setSize(300, 100);
+    loadingDialog.setLocationRelativeTo(inputFrame);
+
+    SwingWorker<List<Patrimoine>, Void> worker =
+        new SwingWorker<>() {
+          @Override
+          protected List<Patrimoine> doInBackground() {
+            var ids = extractInputIds(linksData);
+            List<NamedSnippet> codePatrimoinesVisualisables = new ArrayList<>();
+            for (var id : ids) {
+              var code = googleApi.readDocsContent(authDetails, String.valueOf(id.url()));
+              NamedSnippet namedSnippet = new NamedSnippet(id.name(), code);
+              codePatrimoinesVisualisables.add(namedSnippet);
+            }
+            List<Patrimoine> patrimoinesVisualisables = new ArrayList<>();
+            PatrimoineCompiler patrimoineCompiler = new PatrimoineCompiler();
+            for (NamedSnippet codePatrimoine : codePatrimoinesVisualisables) {
+              String className = new ClassNameExtractor().apply(codePatrimoine.snippet());
+
+              Patrimoine patrimoineVisualisable =
+                  patrimoineCompiler.apply(className, codePatrimoine.snippet());
+              patrimoinesVisualisables.add(patrimoineVisualisable);
+            }
+            return patrimoinesVisualisables;
```

Comment:
put this code in a separate method named compilePatrimoines

Reply:
done